src/Logging/RpcLogger.cs

```diff
@@ -12,34 +12,33 @@
 
 namespace Microsoft.Azure.Functions.PowerShellWorker.Utility
 {
-    internal class RpcLogger : ILogger, IDisposable
+    internal class RpcLogger : ILogger
     {
         private const string SystemLogPrefix = "LanguageWorkerConsoleLog";
         private readonly MessagingStream _msgStream;
         private readonly StringBuilder _systemLogMsg;
         private string _invocationId;
         private string _requestId;
 
-        public RpcLogger(MessagingStream msgStream)
+        internal RpcLogger(MessagingStream msgStream)
         {
             _msgStream = msgStream;
             _systemLogMsg = new StringBuilder();
         }
 
-        public IDisposable BeginScope(string requestId, string invocationId)
+        public void SetContext(string requestId, string invocationId)
         {
             _requestId = requestId;
             _invocationId = invocationId;
-            return this;
         }
 
-        public void Dispose()
+        public void ResetContext()
         {
             _requestId = null;
             _invocationId = null;
         }
 
-        public async void Log(LogLevel logLevel, string message, Exception exception = null, bool isUserLog = false)
+        public void Log(LogLevel logLevel, string message, Exception exception = null, bool isUserLog = false)
```

Comment:
did we want to remove the concept of a "system log" and get rid of the `bool isUserLog = false`?

Reply:
We are still using the system log in case that a profile.ps1 is not found. That is fine because the processing of profile.ps1 only happens once per `PowerShellManager`.
Besides, we need to talk to AzF team (@pragnagopa) to understand the cause, as I do think it's desirable to have system logging ...

Reply:
sounds good